lib/internal/modules/esm/load.js

```diff
@@ -127,19 +127,24 @@ async function defaultLoad(url, context = kEmptyObject) {
 
   throwIfUnsupportedURLScheme(urlInstance, experimentalNetworkImports);
 
-  format ??= await defaultGetFormat(urlInstance, context);
-
-  validateAttributes(url, format, importAttributes);
-
-  if (
-    format === 'builtin' ||
-    format === 'commonjs'
-  ) {
+  if (urlInstance.protocol === 'node:') {
     source = null;
   } else if (source == null) {
     ({ responseURL, source } = await getSource(urlInstance, context));
+    context.source = source;
+  }
+
+  if (format == null || format === 'commonjs') {
+    // Now that we have the source for the module, run `defaultGetFormat` again in case we detect ESM syntax.
+    format = await defaultGetFormat(urlInstance, context);
+  }
```

Comment:
Why do we try to detect ESM if the `resolve` hook hinted that it was CJS? It looks like a mistake, we should try to detect only if the format is ambiguous, no?